src/bridging/providers/across/createAcrossDepositCall.ts

```diff
@@ -0,0 +1,106 @@
+import { CommandFlags, createWeirollDelegateCall } from '../../../weiroll'
+import { ACROSS_MATH_ABI, ACROSS_SPOKE_POOL_ABI } from './abi'
+import { TargetChainId } from '../../../chains/types'
+import { CowShedSdk } from '../../../cow-shed'
+import { AcrossQuoteResult } from './AcrossBridgeProvider'
+import { QuoteBridgeRequest } from '../../types'
+
+const ERC20_BALANCE_OF_ABI = ['function balanceOf(address account) external view returns (uint256)'] as const
+const ERC20_APPROVE_OF_ABI = ['function approve(address spender, uint256 amount) external returns (bool)'] as const
+
+function getSpookPoolContract(sellTokenChainId: TargetChainId): WeirollContract {
+  const spokePoolAddress = ACROSS_SPOOK_CONTRACT_ADDRESSES[sellTokenChainId]
+  if (!spokePoolAddress) {
+    throw new Error('Spoke pool address not found for chain: ' + sellTokenChainId)
+  }
+  return WeirollContract.createContract(new EthersContract(spokePoolAddress, ACROSS_SPOKE_POOL_ABI), CommandFlags.CALL)
+}
+
+function getMathContract(sellTokenChainId: TargetChainId): WeirollContract {
+  const mathContractAddress = ACROSS_MATH_CONTRACT_ADDRESSES[sellTokenChainId]
+  if (!mathContractAddress) {
+    throw new Error('Math contract address not found for chain: ' + sellTokenChainId)
+  }
+
+  return WeirollContract.createContract(new EthersContract(mathContractAddress, ACROSS_MATH_ABI), CommandFlags.CALL)
+}
+
+function getBalanceOfSellTokenContract(sellTokenAddress: string): WeirollContract {
+  return WeirollContract.createContract(
+    new EthersContract(sellTokenAddress, ERC20_BALANCE_OF_ABI),
+    CommandFlags.STATICCALL
+  )
+}
+
+function getApproveSellTokenContract(sellTokenAddress: string): WeirollContract {
+  return WeirollContract.createContract(new EthersContract(sellTokenAddress, ERC20_APPROVE_OF_ABI), CommandFlags.CALL)
+}
+
+export function createAcrossDepositCall(params: {
+  request: QuoteBridgeRequest
+  quote: AcrossQuoteResult
+  cowShedSdk: CowShedSdk
+}): EvmCall {
+  const { quote, cowShedSdk } = params
+  const { sellTokenChainId, sellTokenAddress, buyTokenChainId, buyTokenAddress, owner, recipient } = request
+
+  // Create the relevant weiroll contracts
+  const spokePoolContract = getSpookPoolContract(sellTokenChainId)
+  const mathContract = getMathContract(sellTokenChainId)
+  const balanceOfSellTokenContract = getBalanceOfSellTokenContract(sellTokenAddress)
+  const approveSellTokenContract = getApproveSellTokenContract(sellTokenAddress)
+
+  // Get the cow shed account
+  const cowShedAccount = cowShedSdk.getCowShedAccount(sellTokenChainId, owner)
+
+  const { suggestedFees } = quote
+
+  // Get the weiroll call of the deposit into spoke pool
+  const depositCall = createWeirollDelegateCall((planner) => {
+    // Get bridged amount (balance of the intermediate token at swap time)
+    const sourceAmountIncludingSurplus = planner.add(balanceOfSellTokenContract.balanceOf(cowShedAccount))
```

Comment:
What happens if the user already holds funds in the intermediary token? This will bridge everything, right?

Reply:
Yes, I described this case as one potential issue with this solution. The idea is to not have funds there :) 
Its also what I mentioned about the concurrency problem. Is a known issue, and we have as good to have handling these, but I would not worry at this point